crates/core/component/dex/src/component/action_handler/swap.rs

```diff
@@ -34,32 +34,39 @@ impl ActionHandler for Swap {
     }
 
     async fn check_and_execute<S: StateWrite>(&self, mut state: S) -> Result<()> {
-        let swap_start = std::time::Instant::now();
-        let swap = self;
+        // Only execute the swap if the dex is enabled in the dex params.
+        let dex_params = state.get_dex_params().await?;
 
-        // All swaps will be tallied for the block so the
-        // BatchSwapOutputData for the trading pair/block height can
-        // be set during `end_block`.
-        let mut swap_flow = state.swap_flow(&swap.body.trading_pair);
+        if dex_params.is_enabled {
```

Comment:
would prefer a single:

`ensure!(dex_params.is_enabled, "dex MUST be enabled to process actions");`

This is a nice pattern because it makes the invariant easy to read and will concisely resolve to an early `return Err("... must be ...")` similar to `bail!`.

Reply:
we should also add it to other position actions

Reply:
Oh rad, yeah, that ensure pattern was the kind of suggestion I was looking for, I like that more. :)

However, as for adding it to the other actions, Henry noted that we don't want to disable the "outflow" actions:

```
We do not want to reject the following methods, which are "outflows":

SwapClaim
PositionClose
PositionWithdraw
This ensures that the is_enabled field just turns on / off trading but does not interfere with people getting funds they already have. Otherwise, we'd have a situation where a governance controlled parameter could effectively lock funds.
```

Reply:
for auctions, we will also check this but i can add it later (in addition to auction-level switches)

Reply:
good point, makes sense